deploifai/cli/application/create.py

```diff
@@ -0,0 +1,316 @@
+        try:
+            with spinner():
+                docker_api.push(repository_uri, tag=tag)
+        except docker.errors.APIError:
+            click.secho("Error while pushing image", fg="red")
+            raise click.Abort()
+        click.echo("Done")
+
+        # Use repository uri + tag as image uri
+        image_uri = repository_uri + ':' + tag
+    else:
+        # Use publicly accessible image (assume it is valid)
+        image_uri = prompt(
+            {
+                "type": "input",
+                "name": "image_uri",
+                "message": "Enter public image name (e.g. nginx)",
+            }
+        )['image_uri']
+        if not image_uri:
+            click.secho("Please provide a valid image name", fg="red")
+            raise click.Abort()
+
+    # Get port
+    if not port:
+        port = prompt(
+            {
+                "type": "number",
+                "min_allowed": 0,
+                "max_allowed": 65535,
+                "default": 80,
+                "name": "port",
+                "message": "Port of the container image"
+            }
+        )['port']
+
+    try:
+        port = int(port) if port else 80
+        if port < 0 or port > 65535:
+            raise ValueError
+    except ValueError:
+        click.secho("Please provide a valid port number", fg="red")
+        raise click.Abort()
+
+    # Create plan
+    try:
+        app_config = context.api.get_application_infrastructure_plans(cloud_provider=cloud_profile.provider)
+    except DeploifaiAPIError:
+        click.secho("Could not find application infrastructure plans", fg="red")
+        raise click.Abort()
+
+    # Get app config
+    chosen_config = prompt(
+        {
+            "type": "list",
+            "name": "config",
+            "message": "Choose a size for the application",
+            "choices": [
+                {
+                    "name": "{plan} ({config_type}) - {provider}".format(
+                        plan=config["plan"],
+                        config_type=config["config"],
+                        provider=cloud_profile.provider,
+                    ),
+                    "value": config,
+                }
+                for config in app_config
+            ],
+        }
+    )['config']
+    config = {'plan': chosen_config['plan']}
+
+    # Create application
+    click.echo(f"Creating application with image {image_uri}... ", nl=False)
+    try:
+        with spinner():
+            application = context.api.create_application(project_id, name, cloud_profile.id, config, image_uri, port, formatted_env)
+    except DeploifaiAPIError:
+        click.secho("\nError while creating application", fg="red")
+        raise click.Abort()
+    click.secho(f"\nApplication {application['name']} created successfully", fg="green")
+
+    # Poll for deployment status every 10 seconds
+    click.echo("Checking deployment status (this will take some time)... ", nl=False)
```

Comment:
`"Deploying application (this will take a few minutes, and it's safe to CTRL+C) ..."`